### PR TITLE
[develop < T0067] Sync push stream query modules

### PR DESCRIPTION
**Pull request by niko4299**
### Description

In this PR, the query modules for Kafka streams and Power BI have been put to gqlalchemy to sync if the user needs them.

### Pull request type

Please delete options that are not relevant.

- [X] Feature

Closes (link to issue)

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

######################################

### Reviewer checklist (the reviewer checks this part)
- [ ] Core feature implementation
- [ ] Tests
- [ ] Code documentation
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)

######################################

### CODE REVIEW

File: gqlalchemy/memgraph.py [hunk 1]
```diff
@@ -567,3 +566,29 @@ def get_procedures(self, starts_with: Optional[str] = None, update: bool = False
...
         )
+
+    def with_kafka_stream(self):
+        """Load kafka stream module
+
+        Returns:
+            Memgraph: Memgraph instance
+        """
+        file_text = open('gqlalchemy/query_modules/push_streams/kafka.py','r').read().replace("'", '"')
+        query = f"CALL mg.create_module_file('kafka.py','{file_text}') YIELD *"
```
Comment (Josipmrden) [inline gqlalchemy/memgraph.py:577]: Put this in separate private method with argument of file_path

File: gqlalchemy/memgraph.py [hunk 2]
```diff
@@ -575,3 +578,42 @@ def get_procedures(self, starts_with: Optional[str] = None, update: bool = False
...
+
+    def with_power_bi(self):
+        """Load power_bi stream module
+        Returns:
+            Memgraph: Memgraph instance
+        """
+        file_path = "gqlalchemy/query_modules/push_streams/power_bi.py"
+        module_name = "power_bi_stream.py"
+
+        return self._add_query_module(file_path=file_path, module_name=module_name)
```
Comment (katarinasupe) [inline gqlalchemy/memgraph.py:619]: Add return types

**Review by katarinasupe — CHANGES_REQUESTED** [review]

File: tests/memgraph/test_query_modules.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import pytest
+from gqlalchemy import Memgraph
+from gqlalchemy.exceptions import GQLAlchemyFileNotFoundError
+
```
Comment (brunos252) [inline tests/memgraph/test_query_modules.py:4]: Name of this file might be a bit confusing, there is a `test_query_modules.py` within `tests/graph_algorithms/`, and this tests the addition of query modules, so these tests could either be in `test_memgraph.py`, or be named more specifically, something like `test_add_query_modules.py`

**Review by brunos252 — CHANGES_REQUESTED** [review]
I see you merged some other PR's into your branch. Now when I go to Files changed I can see all those changes that you didn't make, making it a bit harder to review. 
I think since you branched from develop you should only track and merge changes in develop.
If there is a good reason for merging or you think differently, let me know, also regarding any of my comments, as I am also learning here.

File: gqlalchemy/exceptions.py [hunk 4]
```diff
@@ -147,3 +159,40 @@ def __init__(self):
...
     def __init__(self):
         super().__init__(clause=QueryClause.SET)
+
+
+class GQLAlchemyTooLargeTupleInResultQuery(GQLAlchemyError):
+    def __init__(self, clause) -> None:
+        self.message = TOO_LARGE_TUPLE_IN_RESULT_QUERY.format(clause=clause)
+
+
+class GQLAlchemyResultQueryTypeError(TypeError):
```
[the remark this answers is not in the document]
Comment (Josipmrden) [reply gqlalchemy/exceptions.py]: Will fix

File: gqlalchemy/exceptions.py [hunk 5]
```diff
@@ -147,3 +159,40 @@ def __init__(self):
...
+        self.message = INSTANTIATION_ERROR.format(class_name=class_name)
+
+
+class GQLAlchemyDatabaseError(GQLAlchemyError):
+    def __init__(self, message):
+        super().__init__()
+        self.message = message
+
+
+def gqlalchemy_error_handler(func):
```
[the remark this answers is not in the document]
Comment (Josipmrden) [reply gqlalchemy/exceptions.py]: Makes sense

File: gqlalchemy/memgraph.py [hunk 6]
```diff
@@ -16,14 +16,14 @@
 import sqlite3
 
 from typing import Any, Dict, Iterator, List, Optional, Union
-
```
[the remark this answers is not in the document]
Comment (Josipmrden) [reply gqlalchemy/memgraph.py:19]: Will do

File: gqlalchemy/memgraph.py [hunk 7]
```diff
@@ -575,3 +578,42 @@ def get_procedures(self, starts_with: Optional[str] = None, update: bool = False
             if starts_with is None
             else [q for q in self.query_modules if q.name.startswith(starts_with)]
         )
+
+    def _add_query_module(self, file_path: str, module_name: str):
+        """Function for adding query module.
```
[the remark this answers is not in the document]
Comment (Josipmrden) [reply gqlalchemy/memgraph.py:583]: Yes definitely makes sense, will update that the query modules are for Python

**Review by katarinasupe — APPROVED** [review]

**Review by brunos252 — APPROVED** [review]

**Pull request merged**